iroh-net/src/endpoint.rs

```diff
@@ -142,6 +149,26 @@ impl Builder {
 
     // # The very common methods everyone basically needs.
 
+    /// Set the IPv4 bind address.
+    ///
+    /// Setting the port to `0` will use a random port.
+    ///
+    /// By default will use `0.0.0.0:0` to bind to.
+    pub fn bind_addr_v4(mut self, addr: SocketAddrV4) -> Self {
+        self.addr_v4.replace(addr);
+        self
+    }
+
+    /// Set the IPv6 bind address.
+    ///
+    /// Setting the port to `0` will use a random port.
+    ///
+    /// By default will use `::1:0` to bind to.
```

Comment:
`::1` can not and should not be the default, it should be `::`.  `::1` is the loopback address, `::` is the unspecified address.

btw, the way to spell this is `[::]:0`

Reply:
ah right, I can never remember these